segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/rolling/RollingStorage.java

```diff
@@ -279,8 +279,9 @@ public SegmentProperties create(String segmentName, SegmentRollingPolicy rolling
             throw ex;
         }
 
+        val handle = openHandle(segmentName, false);
```

Comment:
Look at line 264. Save the `new RollingSegmentHandle` in a variable and return that. It should contain the same info without having to read the data we've just written.

Reply:
I tried that before , however that generates java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException.   So there is a little bit extra that needs to happen before returning. I'm working on fixing this.